### DISCUSSION TITLE
Include a link to nftexplorer.app next to each NFT ...

### DISCUSSION THREAD

**Discussion by moloch10** [asastats/channel#82] [Ideas]
similar to how each asset includes a link to Tinychart.

![image](https://user-images.githubusercontent.com/97535612/168903409-59c72186-cae3-45f0-8097-e4ac8c8fc289.png)

The details on that site are excellent and I find it to be equivalent to Tinychart.

What are your thoughts?

**Comment by ipaleka** [comment]
The main problem is that we are primarily a business and a business interest is prioritized over the users' interest.

First of all, linking them right now will expose shortcomings of our Alpha version implemented.

Also, we don't know yet does our business model overlap with theirs.

The most important issue here is that afaic the far worst thing we can do is to link them now and remove their links in the future.

So, what would be some benefits for our business to link them?

**Reply by moloch10** [reply]
I understand what you're saying. 

As I mentioned, I see that page as an NFT equivalent to Tinychart; both show historical data which is what I'm after. Since a link to Tiny is there ... you get my point.

If you feel a link to them will be a detriment to Stats, then no link.

Not sure if this should be a separate discussion or not, but what do you think about placing a little copy icon next to each asset id; similar to how addresses have them?

**Reply by ipaleka** [reply]
> As I mentioned, I see that page as an NFT equivalent to Tinychart; both show historical data which is what I'm after. Since a link to Tiny is there ... you get my point.

Let me first explain the difference between regular ASAs and NFTs from our perspective.

We don't record ASA historical data as we don't have any usage from such a database. The only reason to do that would be our users' demands. Still, we deliberately choose to rather sponsor third-party developers in developing and populating such databases and render back the data in aggregate forms to our users in the form of user widgets.

The moment we would have such a solution ready, we'll move Tinychart icon link presentation behavior under user settings, and we'll have a discussion about whether they will be presented by default or not.

With the NFTs is a completely different story: we do collect and keep all the historic data for all the NFTs on Algorand. The reason for that is that we will need those data in order to fill our price-defining algorithm.

Whether we would provide those data to our API users or not is yet to be decided as that's not the core functionality of the ASA Stats website - our goal is to provide API for all the functionalities the website has implemented. We'll provide the final product of those data to our users, of course, and that final product is an NFT price.

Nevertheless, afaic we'll never implement those graphs by ourselves from our data. If we find out we need them then we'd add them to our API and we'd sponsor third parties to develop them in the form of user widgets.

> If you feel a link to them will be a detriment to Stats, then no link.

What I feel is only a cornerstone for my motivation and argumentation in the related discussion. My feelings would never decide on something out of the discussion. The only exception to this (and it's not really an exception), when I **know** something is meaningless and then I, in front of the Team, decide about it without a discussion. And I avoid doing that if someone involved isn't on the same track as me - in such a case I open a discussion and I bring my arguments to it.
 
> Not sure if this should be a separate discussion or not, but what do you think about placing a little copy icon next to each asset id; similar to how addresses have them?

Sorry, I don't understand this, can you explain?

**Reply by moloch10** [reply]
> With the NFTs is a completely different story: we do collect and keep all the historic data for all the NFTs on Algorand. The reason for that is that we will need those data in order to fill our price-defining algorithm.

I was thinking about this very thing, historical data. I know the policy is to not keep it (for standard ASAs) but I wasn't aware that is being saved for NFTs. This completely changes things and it makes perfect sense to not include the link because of business overlap.

> > Not sure if this should be a separate discussion or not, but what do you think about placing a little copy icon next to each asset id; similar to how addresses have them?
> 
> Sorry, I don't understand this, can you explain?

Like this:
![image](https://user-images.githubusercontent.com/97535612/169100021-5c87f8b6-6938-4b5e-b959-f7a4ae9abf6a.png)

Clicking the copy icon would put that ASA's ID in the clipboard.

**Reply by ipaleka** [reply]
> > > Not sure if this should be a separate discussion or not, but what do you think about placing a little copy icon next to each asset id; similar to how addresses have them?
> > 
> > 
> > Sorry, I don't understand this, can you explain?
> 
> Like this: ![image](https://user-images.githubusercontent.com/97535612/169100021-5c87f8b6-6938-4b5e-b959-f7a4ae9abf6a.png)
> 
> Clicking the copy icon would put that ASA's ID in the clipboard.

Oh, I see, yes, that's a nice feature, feel free to create a feature request for it. :)

I was confused by "similar to how addresses have them", what do you mean by that?

**Reply by moloch10** [reply]
> Oh, I see, yes, that's a nice feature, feel free to create a feature request for it. :)

Excellent. Will do. 

> I was confused by "similar to how addresses have them", what do you mean by that?

Sorry, I should have been clearer. Similar to how dapps list a wallet address and usually have a copy icon.

Algoexplorer
![image](https://user-images.githubusercontent.com/97535612/169121154-84309552-5816-4de6-8d01-bf60905bb55a.png)

Algofi
![image](https://user-images.githubusercontent.com/97535612/169121334-bd08ba58-f7cf-47a7-84b4-3618e1d78f6c.png)

**Reply by ipaleka** [reply]
> Sorry, I should have been clearer. Similar to how dapps list a wallet address and usually have a copy icon.

No problem, I thought maybe some browser extension does that for you on our website. :)
